### ISSUE TITLE
Karaf not starting up when webSpoon deployed in a Docker container

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#71]
The following error message occurs when starting webSpoon, but not in every Docker environment.
http://jira.pentaho.com/browse/PDI-16439 reports the same error message.

```
25-Jul-2017 22:02:04.218 INFO [Thread-9] org.apache.karaf.main.Main$KarafLockCallback.lockAquired Lock acquired. Setting startlevel to 100
Warning: Could not get charToByteConverterClass!
22:02:09,977 ERROR [BlueprintURLHandler] Error opening blueprint xml url
java.lang.UnsupportedOperationException: This Transformer, "org.apache.xalan.transformer.TransformerIdentityImpl", does not support the reset functionality.  Specification "null" version "null"
	at javax.xml.transform.Transformer.reset(Transformer.java:82)
	at org.apache.karaf.util.XmlUtils.transform(XmlUtils.java:72)
	at org.apache.karaf.deployer.blueprint.BlueprintTransformer.transform(BlueprintTransformer.java:109)
	at org.apache.karaf.deployer.blueprint.BlueprintURLHandler$Connection.getInputStream(BlueprintURLHandler.java:76)
	at java.net.URL.openStream(URL.java:1045)
	at org.apache.karaf.features.internal.BundleManager.getInputStreamForBundle(BundleManager.java:185)
	at org.apache.karaf.features.internal.BundleManager.doInstallBundleIfNeeded(BundleManager.java:95)
	at org.apache.karaf.features.internal.BundleManager.installBundleIfNeeded(BundleManager.java:89)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.doInstallFeature(FeaturesServiceImpl.java:556)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.installFeatureDependency(FeaturesServiceImpl.java:613)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.doInstallFeature(FeaturesServiceImpl.java:547)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.installFeatures(FeaturesServiceImpl.java:419)
	at org.apache.karaf.features.internal.BootFeaturesInstaller.installBootFeatures(BootFeaturesInstaller.java:92)
	at org.apache.karaf.features.internal.BootFeaturesInstaller$1.run(BootFeaturesInstaller.java:71)
22:03:47,634 ERROR [KarafLifecycleListener] The Kettle Karaf Lifecycle Listener failed to execute properly after waiting for 100 seconds. Releasing lifecycle hold, but some services may be unavailable.
```

**Comment by HiromuHota** [comment]
https://issues.apache.org/jira/browse/XALANJ-2395 reports that this issue has been resolved in 2.7.1.
Upgrading xalan from 2.6.0 to 2.7.1 got me the following:

```
Exception in thread "Thread-15" java.lang.NoClassDefFoundError: org/apache/xml/serializer/ExtendedContentHandler
	at org.apache.xalan.processor.XSLTSchema.build(XSLTSchema.java:325)
	at org.apache.xalan.processor.XSLTSchema.<init>(XSLTSchema.java:72)
	at org.apache.xalan.processor.StylesheetHandler.<init>(StylesheetHandler.java:1290)
	at org.apache.xalan.processor.TransformerFactoryImpl.newTemplatesHandler(TransformerFactoryImpl.java:376)
	at org.apache.xalan.processor.TransformerFactoryImpl.newTemplates(TransformerFactoryImpl.java:867)
	at org.apache.xalan.processor.TransformerFactoryImpl.newTransformer(TransformerFactoryImpl.java:776)
	at org.apache.karaf.util.XmlUtils.transformer(XmlUtils.java:116)
	at org.apache.karaf.util.XmlUtils.transform(XmlUtils.java:77)
	at org.apache.karaf.deployer.blueprint.BlueprintTransformer.analyze(BlueprintTransformer.java:120)
	at org.apache.karaf.deployer.blueprint.BlueprintTransformer.transform(BlueprintTransformer.java:69)
	at org.apache.karaf.deployer.blueprint.BlueprintURLHandler$Connection.getInputStream(BlueprintURLHandler.java:76)
	at java.net.URL.openStream(URL.java:1045)
	at org.apache.karaf.features.internal.BundleManager.getInputStreamForBundle(BundleManager.java:185)
	at org.apache.karaf.features.internal.BundleManager.doInstallBundleIfNeeded(BundleManager.java:95)
	at org.apache.karaf.features.internal.BundleManager.installBundleIfNeeded(BundleManager.java:89)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.doInstallFeature(FeaturesServiceImpl.java:556)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.installFeatureDependency(FeaturesServiceImpl.java:613)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.doInstallFeature(FeaturesServiceImpl.java:547)
	at org.apache.karaf.features.internal.FeaturesServiceImpl.installFeatures(FeaturesServiceImpl.java:419)
	at org.apache.karaf.features.internal.BootFeaturesInstaller.installBootFeatures(BootFeaturesInstaller.java:92)
	at org.apache.karaf.features.internal.BootFeaturesInstaller$1.run(BootFeaturesInstaller.java:71)
Caused by: java.lang.ClassNotFoundException: org.apache.xml.serializer.ExtendedContentHandler
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1285)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1119)
	... 21 more
```

Adding `serializer-2.7.1.jar` as it is depended by `xalan-2.7.1.jar` resolved the issue.

**Comment by HiromuHota** [comment]
xalan=2.6.0 comes from `org="org.pentaho.reporting.library"        name="libfonts"`
https://github.com/pentaho/pentaho-reporting/blob/master/pom.xml

**Comment by HiromuHota** [comment]
Resolved by 96194a6d7c856427d91717599434f812b9261edc
